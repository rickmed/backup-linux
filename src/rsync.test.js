const {__rsync} = require('./rsync')

const PATH = '/some/path'
const PATTERNS = ["/proc/*", "/sys/*"]

it('calls spawn with the correct params', () => {
  const _spawn = jest.fn()
  __rsync(_spawn)(PATH, PATH, PATTERNS)
  expect(_spawn.mock.calls).toMatchSnapshot()
});

it('calls spawn with the correct params of no patterns are passed', () => {
  const _spawn = jest.fn()
  __rsync(_spawn)(PATH, PATH, undefined)
  expect(_spawn.mock.calls).toMatchSnapshot()
});

it('should return the child_process', () => {
  const _spawn = jest.fn( () => 'test')
  const res = __rsync(_spawn)(PATH, PATH, PATTERNS)
  expect(res).toBe('test')
});
