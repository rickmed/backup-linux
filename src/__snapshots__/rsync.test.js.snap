exports[`test calls spawn with the correct params 1`] = `
Array [
  Array [
    "rsync",
    Array [
      "--exclude=/proc/*",
      "--exclude=/sys/*",
      "--info=progress2",
      "-v",
      "--delete",
      "-aAXHt",
      "/some/path",
      "/some/path",
    ],
  ],
]
`;

exports[`test calls spawn with the correct params of no patterns are passed 1`] = `
Array [
  Array [
    "rsync",
    Array [
      "--exclude=/dev/*",
      "--exclude=/proc/*",
      "--exclude=/sys/*",
      "--exclude=/tmp/*",
      "--exclude=/run/*",
      "--exclude=/mnt/*",
      "--exclude=/media/*",
      "--exclude=/var/run/*",
      "--exclude=/var/lock/*",
      "--exclude=/var/cache/apt/archives/*",
      "--exclude=/lost+found",
      "--exclude=/home/*/.gvfs",
      "--exclude=/home/*/.cache",
      "--exclude=/home/*/.local/share/Trash",
      "--exclude=/home/*/.thumbnails/*",
      "--info=progress2",
      "-v",
      "--delete",
      "-aAXHt",
      "/some/path",
      "/some/path",
    ],
  ],
]
`;
